server/game/cards/08.4-FotOG/Yunkai.js

```diff
@@ -0,0 +1,55 @@
+const _ = require('underscore');
+
+const DrawCard = require('../../drawcard.js');
+
+class Yunkai extends DrawCard {
+    setupCardAbilities(ability) {
+        this.action({
+            title: 'Remove characters from challenge',
+            condition: () => this.game.currentChallenge && this.game.currentChallenge.attackingPlayer === this.controller &&
+                             this.game.currentChallenge.getNumberOfParticipants() > 0,
+            cost: [
+                ability.costs.kneelSelf(),
+                ability.costs.discardXGold(() => this.getExtremeParticipatingStr(true), () => this.getExtremeParticipatingStr(false))
```

Comment:
Stylistically, I think having two different methods with descriptive names would be more readable than a boolean flag parameter here. Both methods could use a third method internally that gets the participating strengths to reduce duplication.